sklearn/utils/validation.py

```diff
@@ -8,38 +8,22 @@
 # License: BSD 3 clause
 import warnings
 import numbers
+from inspect import getargspec
 
 import numpy as np
 import scipy.sparse as sp
 
 from ..externals import six
-from inspect import getargspec
-
-FLOAT_DTYPES = (np.float64, np.float32, np.float16)
+from ..exceptions import (DataConversionWarning, EfficiencyWarning,
+                          NotFittedError)
 
 
-class DataConversionWarning(UserWarning):
-    """A warning on implicit data conversions happening in the code"""
-    pass
+FLOAT_DTYPES = (np.float64, np.float32, np.float16)
 
 warnings.simplefilter("always", DataConversionWarning)
-
-
-class NonBLASDotWarning(UserWarning):
-    """A warning on implicit dispatch to numpy.dot"""
-
-
-class NotFittedError(ValueError, AttributeError):
-    """Exception class to raise if estimator is used before fitting
-
-    This class inherits from both ValueError and AttributeError to help with
-    exception handling and backward compatibility.
-    """
-
-
 # Silenced by default to reduce verbosity. Turn on at runtime for
 # performance profiling.
-warnings.simplefilter('ignore', NonBLASDotWarning)
+warnings.simplefilter('ignore', EfficiencyWarning)
```

Comment:
I wonder if we should still silence this by default. Or alternatively we should re-introduce `NonBLASDotWarning` as a sublcass of `EfficiencyWarning` and only silence `NonBLASDotWarning` while keeping `EfficiencyWarning` on by default.